integrationtests/src/test/java/io/kroxylicious/proxy/KrpcFilterIT.java

```diff
@@ -177,19 +167,9 @@ public void shouldModifyProduceMessage() throws Exception {
                         new NewTopic(TOPIC_2, 1, (short) 1))).all().get();
             }
 
-            String config = """
-                    proxy:
-                      address: %s
-                    clusters:
-                      demo:
-                        bootstrap_servers: %s
-                    filters:
-                    - type: ApiVersions
-                    - type: BrokerAddress
-                    - type: ProduceRequestTransformation
-                      config:
-                        transformation: %s
-                    """.formatted(proxyAddress, cluster.getBootstrapServers(), TestEncoder.class.getName());
+            String config = baseConfigBuilder(proxyAddress, cluster.getBootstrapServers())
+                    .addFilter("ProduceRequestTransformation", "transformation", TestEncoder.class.getName())
```

Comment:
I think there is probably a nicer API around adding configuration to filters but I'm happy to leave exploring that for a later PR.